server/api/user/user.controller.ts

```diff
@@ -189,6 +190,37 @@ class UserController {
     res.json(removedUser);
   }
 
+  async removeUser(req: ApiRequest<OrgEdipiParams>, res: Response) {
+    if (!req.appOrg) {
+      throw new NotFoundError('Organization was not found.');
+    }
+
+    const userEDIPI = req.params.edipi;
+
+    const user = await User.findOne({
+      relations: ['roles'],
+      where: {
+        edipi: userEDIPI,
+      },
+      join: {
+        alias: 'user',
+        leftJoinAndSelect: {
+          roles: 'user.roles',
+          org: 'roles.org',
+        },
+      },
+    });
+
+    const roleIndex = (user?.roles ?? []).findIndex((userRole) => userRole.org!.id === req.appOrg!.id);
```

Comment:
I would throw a not found error if the user is `undefined`.  This would allow you to remove the `?` and `!` when accessing it and add some extra safety to the endpoint.

Reply:
I could see value in doing it that way. The only push back I have is that the REST standard states that DELETE operations should be idempotent. I tend to find it more painful to diverge from the standard as the app grows and becomes more robust. For example, the axios-retry library adheres to the standard and will therefore attempt to retry transient failures for DELETE operations. How set are you on your recommended change? Maybe we should discuss on Slack/Meet?